pkg/controller/servicebindingrequest/watch.go

```diff
@@ -0,0 +1,50 @@
+package servicebindingrequest
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// CSVToWatcherMapper creates a EventHandler interface to map ClusterServiceVersion objects back to
+// controller and add given GVK to watch list.
+type CSVToWatcherMapper struct {
+	controller *SBRController
+}
+
+// Map requests directed to CSV objects and exctract related GVK to trigger another watch on
+// controller instance.
+func (c *CSVToWatcherMapper) Map(obj handler.MapObject) []reconcile.Request {
+	olm := NewOLM(c.controller.Client, obj.Meta.GetNamespace())
+	namespacedName := types.NamespacedName{
+		Namespace: obj.Meta.GetNamespace(),
+		Name:      obj.Meta.GetName(),
+	}
+
+	logger := log.WithName("CSVToWatcherMapper").WithValues("Obj.NamespacedName", namespacedName)
+
+	gvks, err := olm.ListGVKsFromCSVNamespacedName(namespacedName)
+	if err != nil {
+		logger.Error(err, "Failed on listing GVK with namespaced-name!")
+		return []reconcile.Request{}
+	}
+
+	for _, gvk := range gvks {
+		logger = logger.WithValues("GVK", gvk)
+		logger.Info("Adding watch for GVK")
+		err = c.controller.AddWatchForGVK(gvk)
```

Comment:
This line could potentially end up adding multiple watches on the same CSV's CRDs. We need to have an in-memory cache to store _which CRDs are we already watching_ .

#### Why
Copies of the same CSV is created in all namespaces when an OLM operator is installed. So technically if there's one operator being installed, there would be 50 new CSVs if there are 50 namespaces in the cluster - that's because CSVs get copied over to every namespace.

Reply:
@sbose78 the controller has a `map` to act as cache, so inside `AddWatchForGVK` you should see that condition being applied.